var/spack/repos/builtin/packages/regcm/package.py

```diff
@@ -56,20 +63,54 @@ def flag_handler(self, name, flags):
         return (None, None, flags)
 
     def configure_args(self):
+        if self.spec.satisfies(r'@4.7.0 %pgi'):
+            raise InstallError('Intel compiler not working with this specific '
+                               'version of RegCM (generates a bug at '
+                               'runtime): please install a newer RegCM '
+                               'version or use GCC.')
+
         args = ['--enable-shared']
 
+        optimizations = self.spec.variants['extension'].value
+        first_optim = optimizations[0]
+
+        if first_optim != 'none':
+            if not (self.spec.satisfies(r'%gcc')
+                    or self.spec.satisfies(r'%intel')):
+                # This means the user chose some optimizations on a different
+                # compiler from GCC and Intel, which are the only compiler
+                # supported by RegCM 4.7.x.
+                raise InstallError('Architecture optimizations are available '
+                                   'only for GCC and Intel compilers.')
+
+            if len(optimizations) > 1 and self.spec.satisfies(r'%gcc'):
+                # https://github.com/spack/spack/issues/974
+                raise InstallError('The GCC compiler does not support '
+                                   'multiple architecture optimizations.')
+
+            args += ('--enable-' + ext for ext in optimizations)
```

Comment:
Given that the package follows a very structured naming convention, it should work seamlessly if you say:
```python
args += self.enable_or_disable('extension')
```

Reply:
Does this work with multiple values?
I mean: if the user gives the following spec: `extension=knl,bdw` (which works only on Intel, see discussion above), does `enable_or_disable` produce the following:
`--enable-knl --enable-bdw --disable-nhl --disable-skl`
?

Reply:
Yes, it works also with multiple values.